packages/toolkit/src/query/fetchBaseQuery.ts

```diff
@@ -27,6 +27,12 @@ export interface FetchArgs extends CustomRequestInit {
   validateStatus?: (response: Response, body: any) => boolean
 }
 
+/**
+ * A mini-wrapper that passes arguments straight through to `window.fetch`.
+ * Avoids storing `window.fetch` in a closure, in order to permit mocking/monkey-patching.
+ */
+const defaultFetchFn = (...args: Parameters<typeof window.fetch>) => window.fetch(...args);
```

Comment:
Should we check for `window` here and warn on it? The [docs already mention handling this in an SSR env](https://redux-toolkit.js.org/rtk-query/api/fetchBaseQuery), but it's the only feedback I have.

Reply:
Yeah, that's a good idea. I've put it immediately within the `fetchBaseQuery` creation call though, rather than when a fetch is attempted. Caught quicker, and spams less